portfolio.py

```diff
@@ -2,13 +2,40 @@
 Allocation of the asset.
 """
 
+import numpy as np
+from thousandaire.constants import TRADING_INSTRUMENTS
+
 class Portfolio(dict):
     """
     Set the portfolio
     """
-    def __init__(self, date):
+    def __init__(self, date, target=None, np_array=None):
         dict.__init__(self)
         self.date = date
+        if target is not None:
+            self.decode_nparray(target, np_array)
+
+    def decode_nparray(self, target, np_array):
+        """
+        Decode numpy array into portfolio.
+        """
+        if len(np_array) != len(TRADING_INSTRUMENTS[target]):
+            raise TypeError("Cannot decode into the target type.")
+        for index in range(0, len(np_array), 1):
+            instrument = TRADING_INSTRUMENTS[target][index]
+            self[instrument] = np_array[index]
+
+    def encode_to_nparray(self, target):
+        """
+        Encode the positions into numpy type.
+        """
+        encode_list = list()
+        for instrument in TRADING_INSTRUMENTS[target]:
+            if instrument in self:
+                encode_list.append(self[instrument])
+            else:
+                encode_list.append(0)
```

Comment:
encoded_list = [self.get(instrument, 0) for instrument in TRADING_INSTURMENTS[target]]

Reply:
With this one line code, you can even return it directly: `return np.array(...)`